addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/check-launch-dependency/check-launch-dependency.js

```diff
@@ -0,0 +1,433 @@
+            this.payloadOrConfig.object(inPayloadKeys.requestContext),
+            this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+            this.payloadOrConfig.string(inPayloadKeys.productId),
+            this.payloadOrConfig.string(inPayloadKeys.envTypeId),
+            this.payloadOrConfig.string(inPayloadKeys.envTypeConfigId),
+        ]);
+        const projectId = resolvedVars.projectId;
+        const [envTypeConfigService] = await this.mustFindServices(['envTypeConfigService']);
+        const envTypeConfig = await envTypeConfigService.mustFind(requestContext, envTypeId, { id: envTypeConfigId });
+
+        // Create dynamic namespace follows the existing pattern of namespace
+        resolvedVars.namespace = `analysis-${Date.now()}`;
+        const resolvedInputParams = await this.resolveVarExpressions(envTypeConfig.params, resolvedVars);
+        const templateOutputs = await this.getTemplateOutputs(requestContext, envTypeId, productId);
+        const needsAlb = _.get(templateOutputs.NeedsALB, 'Value', false);
+        const maxAlbWorkspacesCount = _.get(templateOutputs.MaxCountALBDependentWorkspaces, 'Value', MAX_COUNT_ALB_DEPENDENT_WORKSPACES);
+        if (needsAlb) {
+            return await this.provisionAlb(requestContext, resolvedVars, projectId, resolvedInputParams, maxAlbWorkspacesCount);
+        }
+    }
+
+    /**
+    * Method to provision ALB. The method checks if the max count of ALB possible exists and
+    * checks if there is an ALB aready exists for the AWS account and provisions if not exists.
+    *
+    * @param requestContext
+    * @param resolvedVars
+    * @param projectId
+    * @param resolvedInputParams
+    * @param maxAlbWorkspacesCount
+    * @returns {Promise<>}
+    */
+    async provisionAlb(requestContext, resolvedVars, projectId, resolvedInputParams, maxAlbWorkspacesCount) {
+        const [ALBService] = await this.mustFindServices(['albService']);
+        const count = await ALBService.albDependentWorkspacesCount(requestContext, projectId);
+        const albExists = await ALBService.checkAlbExists(requestContext, projectId);
+        if (count >= maxAlbWorkspacesCount) {
+            throw new Error(`Error provisioning environment. Reason: Maximum workspaces using ALB has reached`);
+        }
+        if (albExists) {
+            this.print({
+                msg: `ALB Already exists for the Account. Skipping ALB Creation`,
+            });
+        } else {
+            this.print({
+                msg: `Workspace needs ALB. Provisioning an ALB.`,
+            });
+            const stackInput = await ALBService.getStackCreationInput(requestContext, resolvedVars, resolvedInputParams, projectId);
+            //Storing Dependency type so the stack completion can be handled for different dependencies
+            this.state.setKey('DEPENDENCY_TYPE', 'ALB');
+            //Create Stack
+            return await this.deployStack(requestContext, resolvedVars, stackInput);
```

Comment:
What if ALB doesn't exist but we have one in-flight. We want to avoid deploying this stack in case one is in progress, right?